network/middleware.go

```diff
@@ -48,17 +48,14 @@ type Middleware interface {
 	// Unsubscribe unsubscribes the middleware from a channel.
 	Unsubscribe(channel Channel) error
 
-	// Ping pings the target node and returns the ping RTT or an error
-	Ping(targetID flow.Identifier) (message.PingResponse, time.Duration, error)
-
-	// UpdateAllowList fetches the most recent identity of the nodes from overlay
-	// and updates the underlying libp2p node.
-	UpdateAllowList()
-
 	// UpdateNodeAddresses fetches and updates the addresses of all the staked participants
 	// in the Flow protocol.
 	UpdateNodeAddresses()
 
+	RoutingSystem() routing.Routing
+
+	Host() host.Host
+
```

Comment:
Highlighting this comment

Reply:
867d63c

fdc39c7

Reply:
I did not move the `network.RegisterPingService` call into the Ping Engine, because as mentioned before:
* Every node needs to register a PingService to handle incoming pings. In other words, every node needs to call `network.RegisterPingService`
* Only Access nodes need to start a PingEngine to send outgoing pings. 

Thus, I think it's simpler to initialize the PingService in a single place for all node types, and then Access Nodes can pass this to the PingEngine.